code/game/gamemodes/malfunction/newmalf_ability_trees/HELPERS.dm

```diff
@@ -171,11 +171,37 @@
 /proc/get_unhacked_apcs(var/mob/living/silicon/ai/user)
 	var/list/H = list()
 	for(var/obj/machinery/power/apc/A in machines)
-		if(A.hacker && A.hacker == user)
+		if(!A.hacker && A.hacker != user)
 			continue
 		H.Add(A)
 	return H
 
+// Proc: get_hacked_apcs()
+// Parameters: None
+// Description: Returns a list of all hacked APCs
+/proc/get_hacked_apcs(var/mob/living/silicon/ai/user)
+	var/list/H = list()
+	for(var/obj/machinery/power/apc/A in machines)
+		if(A.hacker || A.hacker == user)
```

Comment:
This is also bad. It _skips_ APCs which are hacked by the user.

Should instead be:
```
if (A.hacker && A.hacker == user)
    H.Add(A)
```